### DISCUSSION TITLE
Very very impressive!

### DISCUSSION THREAD

**Discussion by yaaama** [pierre-rouleau/pel#30] [General]
My biggest compliments.
This repo is a testament of excellence and so much hardwork! 

I'm genuinely so impressed with how vast and comprehensive this repo is! 
I'm especially impressed with the dedication to documentation, which is something that 98% of other developers don't think about/ do very poorly. But the documentation I've read from here is so well done.

Bravo, I appreciate the hard work you've put into this project, its truly inspiring :smile:
Perhaps I might even focus on learning elisp to help out 😆

**Comment by pierre-rouleau** [comment]
Thanks for your great feedback @yaaama!  I started that project as a way to learn Emacs. 

Emacs is interesting in many ways; it's open source, it's old yet still very active and super useful.  Lisp in itself is great and even though Emacs Lisp is not as powerful as other LISPs, it's still extremely powerful.   It opens up on ideas about a lot of things.   

Because Emacs is so vast, it takes time to learn.  My way to cope with it was to write code and document it.   Emacs itself has a extremely powerful documentation system, based on docstrings.   But I wanted a way to quickly access the knowledge and organize my thinking, document what exists in Emacs and what I add.  The only way I found that would make it fast to access, and dense enough to fit lots of info in relatively small space was to create the info in tables.  The tables have a lot of contextual info based on text but also icons and foreground and background colours.  they have lots of hyperlinks too.  They are best used in a browser that render the PDFs.  But they can be launched by commands (keys) from Emacs as well.

Maintaining the tables that generate the PDFs does take time, but it also helps me to find things later on.

But there's a lot to do still.  I have more ideas than time to do them.  If you see something that is not clear or wrong, let me know.

And if you want to propose something, go ahead!

**Reply by pierre-rouleau** [reply]
BTW, @yaaama, you might be interested in reading "A Philosophy of Software Design" by John Ousterhout (TCL designer, among other), where he describes why comment are important in code (see section 13: Comments Should Describe Things that Aren't Obvious from the Code".

**Reply by yaaama** [reply]
I’ll have a look through it, thank you!!
